src/vs/workbench/contrib/scm/browser/dirtydiffDecorator.ts

```diff
@@ -1195,97 +1188,125 @@ export class DirtyDiffModel extends Disposable {
 			}, (err) => onUnexpectedError(err));
 	}
 
-	private setChanges(changes: IChange[]): void {
-		const diff = sortedDiff(this._changes, changes, compareChanges);
+	private setChanges(changes: LabeledChange[]): void {
+		const diff = sortedDiff(this._changes, changes, (a, b) => compareChanges(a.change, b.change));
 		this._changes = changes;
 		this._onDidChange.fire({ changes, diff });
 	}
 
-	private diff(): Promise<IChange[] | null> {
+	private diff(): Promise<LabeledChange[] | null> {
 		return this.progressService.withProgress({ location: ProgressLocation.Scm, delay: 250 }, async () => {
-			return this.getOriginalURIPromise().then(originalURI => {
-				if (this._disposed || this._model.isDisposed() || !originalURI) {
-					return Promise.resolve([]); // disposed
-				}
-
-				if (!this.editorWorkerService.canComputeDirtyDiff(originalURI, this._model.resource)) {
-					return Promise.resolve([]); // Files too large
-				}
+			const originalURIs = await this.getQuickDiffsPromise();
+			if (this._disposed || this._model.isDisposed() || (originalURIs.length === 0)) {
+				return Promise.resolve([]); // disposed
+			}
 
-				const ignoreTrimWhitespaceSetting = this.configurationService.getValue<'true' | 'false' | 'inherit'>('scm.diffDecorationsIgnoreTrimWhitespace');
-				const ignoreTrimWhitespace = ignoreTrimWhitespaceSetting === 'inherit'
-					? this.configurationService.getValue<boolean>('diffEditor.ignoreTrimWhitespace')
-					: ignoreTrimWhitespaceSetting !== 'false';
+			if (originalURIs.some(quickDiff => !this.editorWorkerService.canComputeDirtyDiff(quickDiff.originalResource, this._model.resource))) {
+				return Promise.resolve([]); // Files too large
+			}
```

Comment:
If I understand this correctly, as soon as one of the dirty diffs is too big, none will be rendered. That doesn't seem right.

Reply:
My understanding of `canComputDirtyDiff` based on the comment "Files too large" was that if the file is too large we cannot compute the diff. Since current file is always going to be the same size it seemed safe to bail after the first `false` from `canComputeDirtyDiff`.

Reply:
This seemed practical since usually the original and the current file will likely be similar in size. However, we can also just filter out those that are too large.